### PR TITLE
Add PiKISS-GUI app

### PR DESCRIPTION
**Pull request by Jai-JAP**
PiKISS GUI is a GTK frontend for piKISS created by @krishenriksen and modified by @Jai-JAP for semi-automated app syncs from piKISS.

The modified version of app is located at https://github.com/Jai-JAP/PiKISS-GUI

This app is tested and ready to merge.

### CODE REVIEW

**Comment by Botspot** [comment]
This app is similar to the Katoolin app request, in that the author is long gone. If something breaks in the future, @krishenriksen won't be around to help - he's moved on from open source development AFAICT.

Is someone willing to fork his repository and volunteer to continue maintaining it?

**Comment by Jai-JAP** [comment]
I have forked it and app is already fetching my fork.
I am also planning some more improvements. So this can be added as I will be maintaining the fork.

**Comment by Botspot** [comment]
> I have forked it and app is already fetching my fork. I am also planning some more improvements. So this can be added as I will be maintaining the fork.

Great! Maybe you can improve the Pi-Apps GUI at some point. ;) "announcing YAD 2.0!"

File: apps/PiKISS GUI/uninstall [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+#Allow packages required by this app to be uninstalled
+purge_packages || exit 1
```
Comment (Botspot) [inline apps/PiKISS GUI/uninstall:4]: Your uninstall script does not sufficiently uninstall the app. Unless I'm mistaken, multiple folders will be left over, unless your `./build.sh` deleted them all already.

**Comment by Botspot** [comment]
...And if failed to install. :(
```
OS: Raspbian GNU/Linux 11 (bullseye)
OS architecture: 32-bit
Pi-Apps local commit ID: 6dd1b704bcfd73695e41db21aafe62a50a08cb48
Github latest commit ID: 2a2c6241590935d83cb4e973c7b1dc25904be38f	
Kernel: aarch64 5.10.92-v8+
Device model: Raspberry Pi 4 Model B Rev 1.4
Ram size: 7.76 GB
Raspberry Pi OS image version: 2021-10-30
Language: en_US.UTF-8

BEGINNING OF LOG FILE:
-----------------------

Downloading pikiss-gui repository... Done
Will install these packages: meson gobject-introspection libgee-0.8-dev libgirepository1.0-dev libgtk-3-dev valac 
Running sudo apt update...
Hit:1 https://itai-nelken.github.io/weekly-box86-debs/debian  InRelease
Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease
Hit:3 http://raspbian.raspberrypi.org/raspbian bullseye InRelease
Ign:4 https://download.puffinbrowser.com/repo stable InRelease
Hit:5 https://deb.oxen.io bullseye InRelease
Hit:6 https://adoptopenjdk.jfrog.io/adoptopenjdk/deb bullseye InRelease
Hit:7 https://adoptium.jfrog.io/artifactory/deb bullseye InRelease
Hit:8 https://download.puffinbrowser.com/repo stable Release
All packages are up to date.
apt update complete.
Creating an empty apt-package to install the necessary apt packages...
It will be named: pi-apps-51b4a14d
Depends: gobject-introspection, libgee-0.8-dev, libgirepository1.0-dev, libgtk-3-dev, meson, valac
Installing the pi-apps-51b4a14d package...
The following packages were automatically installed and are no longer required:
  fonts-league-spartan libqt5charts5
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  gir1.2-gee-0.8 libgee-0.8-dev libvala-0.48-0 libvalacodegen-0.48-0 meson
  valac valac-0.48-vapi valac-bin
The following NEW packages will be installed:
  gir1.2-gee-0.8 libgee-0.8-dev libvala-0.48-0 libvalacodegen-0.48-0 meson
  pi-apps-51b4a14d valac valac-0.48-vapi valac-bin
0 upgraded, 9 newly installed, 0 to remove and 0 not upgraded.
Get:1 /home/pi/pi-apps-51b4a14d.deb pi-apps-51b4a14d all 1.0 [792 B]
Get:2 http://raspbian.mirror.constant.com/raspbian bullseye/main armhf gir1.2-gee-0.8 armhf 0.20.4-1 [36.0 kB]
Get:3 http://mirror.pit.teraswitch.com/raspbian/raspbian bullseye/main armhf libgee-0.8-dev armhf 0.20.4-1 [67.1 kB]
Get:4 http://raspbian.mirror.constant.com/raspbian bullseye/main armhf libvala-0.48-0 armhf 0.48.17-1 [721 kB]
Get:5 http://mirror.pit.teraswitch.com/raspbian/raspbian bullseye/main armhf libvalacodegen-0.48-0 armhf 0.48.17-1 [606 kB]
Get:6 http://mirror.pit.teraswitch.com/raspbian/raspbian bullseye/main armhf meson all 0.56.2-1 [437 kB]
Get:7 http://mirror.pit.teraswitch.com/raspbian/raspbian bullseye/main armhf valac-0.48-vapi all 0.48.17-1 [1,003 kB]
Get:8 http://mirror.pit.teraswitch.com/raspbian/raspbian bullseye/main armhf valac-bin armhf 0.48.17-1 [302 kB]
Get:9 http://mirror.pit.teraswitch.com/raspbian/raspbian bullseye/main armhf valac armhf 0.48.17-1 [379 kB]
Fetched 3,550 kB in 3s (1,133 kB/s)
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 185169 files and directories currently installed.)

Unpacking gir1.2-gee-0.8:armhf (0.20.4-1) ...

Unpacking libgee-0.8-dev:armhf (0.20.4-1) ...

Unpacking libvala-0.48-0:armhf (0.48.17-1) ...

Unpacking libvalacodegen-0.48-0:armhf (0.48.17-1) ...

Unpacking meson (0.56.2-1) ...

Unpacking valac-0.48-vapi (0.48.17-1) ...

Unpacking valac-bin (0.48.17-1) ...

Unpacking valac (0.48.17-1) ...

Unpacking pi-apps-51b4a14d (1.0) ...

Apt finished.
Package installation complete.
Installing piKiss...

HJJ
PiKISS
======

Installing at /home/pi/piKiss. Please wait...


WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
The following packages were automatically installed and are no longer required:
  fonts-league-spartan libqt5charts5
Use 'sudo apt autoremove' to remove them.
The following NEW packages will be installed:
  dialog
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 252 kB of archives.
After this operation, 1,020 kB of additional disk space will be used.
Get:1 http://mirror.pit.teraswitch.com/raspbian/raspbian bullseye/main armhf dialog armhf 1.3-20201126-1 [252 kB]
Fetched 252 kB in 1s (292 kB/s)
Selecting previously unselected package dialog.

(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 185713 files and directories currently installed.)

Preparing to unpack .../dialog_1.3-20201126-1_armhf.deb ...

Unpacking dialog (1.3-20201126-1) ...

Setting up dialog (1.3-20201126-1) ...

Processing triggers for man-db (2.9.4-2) ...

Cloning into 'piKiss'...

PiKISS installed!
=================

cd /home/pi/piKiss, type or click ./piKiss.sh. You have an Menu shortcut, too!. Go to:

 · Raspberry Pi OS: Menu > System Tools > PiKISS
 · Twister OS: Menu > Accesories > PiKISS


Installed piKiss successfully.
The Meson build system
Version: 0.56.2
Source dir: /home/pi/pikiss-gui
Build dir: /home/pi/pikiss-gui/build
Build type: native build
Project name: pikiss-gui
Project version: 0.0.8
C compiler for the host machine: cc (gcc 10.2.1 "cc (Raspbian 10.2.1-6+rpi1) 10.2.1 20210110")
C linker for the host machine: cc ld.bfd 2.35.2
Vala compiler for the host machine: valac (valac 0.48.17)
Host machine cpu family: arm
Host machine cpu: arm
Configuring config.h using configuration
Library config found: YES
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Run-time dependency gio-unix-2.0 found: YES 2.66.8
Run-time dependency glib-2.0 found: YES 2.66.8
Run-time dependency gtk+-3.0 found: YES 3.24.24
Run-time dependency gdk-3.0 found: YES 3.24.24
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pango found: YES 1.46.2
Run-time dependency gee-0.8 found: YES 0.20.4
Found CMake: /usr/bin/cmake (3.18.4)
Run-time dependency libgnome-menu-3.0 found: NO (tried pkgconfig and cmake)

meson.build:40:0: ERROR: Dependency "libgnome-menu-3.0" not found, tried pkgconfig and cmake

A full log can be found at /home/pi/pikiss-gui/build/meson-logs/meson-log.txt
ninja: Entering directory `build'
ninja: error: loading 'build.ninja': No such file or directory
ninja: Entering directory `build'
ninja: error: loading 'build.ninja': No such file or directory
These packages were: gobject-introspection, libgee-0.8-dev, libgirepository1.0-dev, libgtk-3-dev, meson, valac
Waiting until APT locks are released... Done

Failed to install PiKISS GUI!
```

**Comment by cycool29** [comment]
> ```
> meson.build:40:0: ERROR: Dependency "libgnome-menu-3.0" not found, tried pkgconfig and cmake
> ```

`sudo apt install libgnome-menu-3-dev` works for me.

**Comment by Jai-JAP** [comment]
Yeah I missed that

**Comment by Botspot** [comment]
> `sudo apt install libgnome-menu-3-dev` works for me.

That worked for me too. Thank you, @cycool29.

@Jai-JAP, could you add that dependency to the list? Also, could you remove the warning that the user may need to type in their password to run `sudo`? Raspberry Pi OS and TwisterOS both use passwordless `sudo`, and on top of that, *all other Pi-Apps scripts* proceed as normal and issue no warnings about this.

File: apps/PiKISS GUI/uninstall (hunk 1 above)
Comment (Jai-JAP) [reply apps/PiKISS GUI/uninstall]: Some commits were not pushed online.

Fixed.

**Comment by Botspot** [comment]
Please remove this.
```
This script may ask you for your password. 
Please enter your password if asked.

Press [ENTER] to continue.
```

**Comment by Jai-JAP** [comment]
Done

**Comment by Botspot** [comment]
Strange. It said "Pikiss apps successfully synced", then later in the output it said "Syncing Apps with PiKISS..."
Is there a difference in these two messages?

**Comment by Jai-JAP** [comment]
> Strange. It said "Pikiss apps successfully synced", then later in the output it said "Syncing Apps with PiKISS..." Is there a difference in these two messages?

Checking it out

**Comment by Jai-JAP** [comment]
Can you send the log

**Comment by Jai-JAP** [comment]
Also send a lod after adding set +x at the top of the script

**Comment by Botspot** [comment]
> Can you send the log

This should be enough.
```
Found ninja-1.10.1 at /usr/bin/ninja
ninja: Entering directory `build'
[1/3] Compiling Vala source ../src/Application.vala
[2/3] Compiling C object pikiss-gui.p/meson-generated_src_Application.c.o
[3/3] Linking target pikiss-gui
ninja: Entering directory `build'
[0/1] Installing files.

PiKISS Apps successfully synced.
?25lWaiting 10 seconds before exiting...
Waiting 9 seconds before exiting...
Waiting 8 seconds before exiting...
Waiting 7 seconds before exiting...
Waiting 6 seconds before exiting...
Waiting 5 seconds before exiting...
Waiting 4 seconds before exiting...
Waiting 3 seconds before exiting...
Waiting 2 seconds before exiting...
Waiting 1 seconds before exiting...
Waiting 0 seconds before exiting...
J
?12;25h


Creating PiKISS GUI category dirs...
PiKISS GUI category dirs created successfully.

Creating PiKISS GUI menuentry...
PiKISS GUI menuentry created successfully.

Creating App import script...
App import script created successfully.

Creating PiKISS GUI Sync Apps menuentry...
PiKISS GUI Sync Apps menuentry created successfully.
For resyncing apps with PiKISS use menu launcher "PiKISS GUI Sync Apps"

Creating uninstall script
Uninstall script created successfully.

Syncing Apps with PiKISS...
```

**Comment by Jai-JAP** [comment]
Can you retry now

**Comment by Botspot** [comment]
> Can you retry now

Sure.

**Comment by Botspot** [comment]
I didn't see a difference. It still had that unusual countdown, and still later said "Syncing Apps with PiKISS..."

**Comment by Jai-JAP** [comment]
Strange building manually doesn't give this error
```
pi@TwisterOS:~/dev/pikiss-gui $ ./build.sh 
The Meson build system
Version: 0.56.1
Source dir: /home/pi/dev/pikiss-gui
Build dir: /home/pi/dev/pikiss-gui/build
Build type: native build
Project name: pikiss-gui
Project version: 0.0.8
C compiler for the host machine: cc (gcc 8.3.0 "cc (Raspbian 8.3.0-6+rpi1) 8.3.0")
C linker for the host machine: cc ld.bfd 2.31.1
Vala compiler for the host machine: valac (valac 0.42.5)
Host machine cpu family: arm
Host machine cpu: armv7l
Configuring config.h using configuration
Library config found: YES
Found pkg-config: /usr/bin/pkg-config (0.29)
Run-time dependency gio-unix-2.0 found: YES 2.58.3
Run-time dependency glib-2.0 found: YES 2.58.3
Run-time dependency gtk+-3.0 found: YES 3.24.5
Run-time dependency gdk-3.0 found: YES 3.24.5
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pango found: YES 1.42.3
Run-time dependency gee-0.8 found: YES 0.20.1
Run-time dependency libgnome-menu-3.0 found: YES 3.31.4
Library m found: YES
Program post_install.py found: YES (/usr/bin/python3 /home/pi/dev/pikiss-gui/post_install.py)
Build targets in project: 1

Found ninja-1.10.1 at /usr/bin/ninja
ninja: Entering directory `build'
[3/3] Linking target pikiss-gui
ninja: Entering directory `build'
[0/1] Installing files.
Installing pikiss-gui to /home/pi/.local/bin
Installing /home/pi/dev/pikiss-gui/data/pikiss-gui.png to /home/pi/.local/share/icons
Running custom install script '/usr/bin/python3 /home/pi/dev/pikiss-gui/post_install.py'



Creating PiKISS GUI category dirs...
PiKISS GUI category dirs created successfully.

Creating PiKISS GUI menuentry...
PiKISS GUI menuentry created successfully.

Creating App import script...
App import script created successfully.

Creating PiKISS GUI Sync Apps menuentry...
PiKISS GUI Sync Apps menuentry created successfully.
For resyncing apps with PiKISS use menu launcher "PiKISS GUI Sync Apps"

Creating uninstall script
Uninstall script created successfully.

Syncing Apps with PiKISS...



PiKISS Apps successfully synced.


```

**Comment by Botspot** [comment]
> Strange building manually doesn't give this error
> 
> ```
> pi@TwisterOS:~/dev/pikiss-gui $ ./build.sh 
> The Meson build system
> Version: 0.56.1
> Source dir: /home/pi/dev/pikiss-gui
> Build dir: /home/pi/dev/pikiss-gui/build
> Build type: native build
> Project name: pikiss-gui
> Project version: 0.0.8
> C compiler for the host machine: cc (gcc 8.3.0 "cc (Raspbian 8.3.0-6+rpi1) 8.3.0")
> C linker for the host machine: cc ld.bfd 2.31.1
> Vala compiler for the host machine: valac (valac 0.42.5)
> Host machine cpu family: arm
> Host machine cpu: armv7l
> Configuring config.h using configuration
> Library config found: YES
> Found pkg-config: /usr/bin/pkg-config (0.29)
> Run-time dependency gio-unix-2.0 found: YES 2.58.3
> Run-time dependency glib-2.0 found: YES 2.58.3
> Run-time dependency gtk+-3.0 found: YES 3.24.5
> Run-time dependency gdk-3.0 found: YES 3.24.5
> Run-time dependency cairo found: YES 1.16.0
> Run-time dependency pango found: YES 1.42.3
> Run-time dependency gee-0.8 found: YES 0.20.1
> Run-time dependency libgnome-menu-3.0 found: YES 3.31.4
> Library m found: YES
> Program post_install.py found: YES (/usr/bin/python3 /home/pi/dev/pikiss-gui/post_install.py)
> Build targets in project: 1
> 
> Found ninja-1.10.1 at /usr/bin/ninja
> ninja: Entering directory `build'
> [3/3] Linking target pikiss-gui
> ninja: Entering directory `build'
> [0/1] Installing files.
> Installing pikiss-gui to /home/pi/.local/bin
> Installing /home/pi/dev/pikiss-gui/data/pikiss-gui.png to /home/pi/.local/share/icons
> Running custom install script '/usr/bin/python3 /home/pi/dev/pikiss-gui/post_install.py'
> 
> 
> 
> Creating PiKISS GUI category dirs...
> PiKISS GUI category dirs created successfully.
> 
> Creating PiKISS GUI menuentry...
> PiKISS GUI menuentry created successfully.
> 
> Creating App import script...
> App import script created successfully.
> 
> Creating PiKISS GUI Sync Apps menuentry...
> PiKISS GUI Sync Apps menuentry created successfully.
> For resyncing apps with PiKISS use menu launcher "PiKISS GUI Sync Apps"
> 
> Creating uninstall script
> Uninstall script created successfully.
> 
> Syncing Apps with PiKISS...
> 
> 
> 
> PiKISS Apps successfully synced.
> ```

Were you watching the terminal the entire time?
I noticed that once the countdown is complete, it erased itself from the terminal, but was still visible in the logfile.

**Comment by cycool29** [comment]
@Jai-JAP You need to remove `/usr/bin/ninja` first if `ninja` is installed before installing `ninja-build` as both of them share the same binary file. 

```bash
if package_installed ninja; then
  sudo rm -f /usr/bin/ninja
  REINSTALL_NINJA=1
fi

# Install PiKISS-GUI...

[ "$REINSTALL_NINJA" ] && sudo apt-get install ninja --reinstall

```

**Comment by Jai-JAP** [comment]
> > Strange building manually doesn't give this error
> > ```
> > pi@TwisterOS:~/dev/pikiss-gui $ ./build.sh 
> > The Meson build system
> > Version: 0.56.1
> > Source dir: /home/pi/dev/pikiss-gui
> > Build dir: /home/pi/dev/pikiss-gui/build
> > Build type: native build
> > Project name: pikiss-gui
> > Project version: 0.0.8
> > C compiler for the host machine: cc (gcc 8.3.0 "cc (Raspbian 8.3.0-6+rpi1) 8.3.0")
> > C linker for the host machine: cc ld.bfd 2.31.1
> > Vala compiler for the host machine: valac (valac 0.42.5)
> > Host machine cpu family: arm
> > Host machine cpu: armv7l
> > Configuring config.h using configuration
> > Library config found: YES
> > Found pkg-config: /usr/bin/pkg-config (0.29)
> > Run-time dependency gio-unix-2.0 found: YES 2.58.3
> > Run-time dependency glib-2.0 found: YES 2.58.3
> > Run-time dependency gtk+-3.0 found: YES 3.24.5
> > Run-time dependency gdk-3.0 found: YES 3.24.5
> > Run-time dependency cairo found: YES 1.16.0
> > Run-time dependency pango found: YES 1.42.3
> > Run-time dependency gee-0.8 found: YES 0.20.1
> > Run-time dependency libgnome-menu-3.0 found: YES 3.31.4
> > Library m found: YES
> > Program post_install.py found: YES (/usr/bin/python3 /home/pi/dev/pikiss-gui/post_install.py)
> > Build targets in project: 1
> > 
> > Found ninja-1.10.1 at /usr/bin/ninja
> > ninja: Entering directory `build'
> > [3/3] Linking target pikiss-gui
> > ninja: Entering directory `build'
> > [0/1] Installing files.
> > Installing pikiss-gui to /home/pi/.local/bin
> > Installing /home/pi/dev/pikiss-gui/data/pikiss-gui.png to /home/pi/.local/share/icons
> > Running custom install script '/usr/bin/python3 /home/pi/dev/pikiss-gui/post_install.py'
> > 
> > 
> > 
> > Creating PiKISS GUI category dirs...
> > PiKISS GUI category dirs created successfully.
> > 
> > Creating PiKISS GUI menuentry...
> > PiKISS GUI menuentry created successfully.
> > 
> > Creating App import script...
> > App import script created successfully.
> > 
> > Creating PiKISS GUI Sync Apps menuentry...
> > PiKISS GUI Sync Apps menuentry created successfully.
> > For resyncing apps with PiKISS use menu launcher "PiKISS GUI Sync Apps"
> > 
> > Creating uninstall script
> > Uninstall script created successfully.
> > 
> > Syncing Apps with PiKISS...
> > 
> > 
> > 
> > PiKISS Apps successfully synced.
> > ```
> 
> Were you watching the terminal the entire time? I noticed that once the countdown is complete, it erased itself from the terminal, but was still visible in the logfile.

yes its intended to disappear from terminal but logfile always catches it

**Comment by Botspot** [comment]
> yes its intended to disappear from terminal but logfile always catches it

Why is a 10-second countdown necessary at all? It seems unnecessary.

**Comment by Jai-JAP** [comment]
Its for users who manually install it

**Comment by Jai-JAP** [comment]
> @Jai-JAP You need to remove `/usr/bin/ninja` first if `ninja` is installed before installing `ninja-build` as both of them share the same binary file.
> 
> ```shell
> if package_installed ninja; then
>   sudo rm -f /usr/bin/ninja
>   REINSTALL_NINJA=1
> fi
> 
> # Install PiKISS-GUI...
> 
> [ "$REINSTALL_NINJA" ] && sudo apt-get install ninja --reinstall
> ```

add this causes hte app not to install at all error log below 

<details>
pi@TwisterOS:~ $ ~/pi-apps/manage install PiKISS\ GUI
Installing PiKISS GUI...

Downloading pikiss-gui repository... Done
Will install these packages: meson gobject-introspection libgee-0.8-dev libgirepository1.0-dev libgtk-3-dev valac libgnome-menu-3-dev 
Running sudo apt update...
Hit:1 https://mcpi-revival.github.io/mcpi-packages buster InRelease
Hit:2 http://archive.raspberrypi.org/debian buster InRelease
Hit:3 http://ppa.launchpad.net/apt-fast/stable/ubuntu focal InRelease
Hit:4 http://raspbian.raspberrypi.org/raspbian buster InRelease
Hit:5 http://ports.ubuntu.com/ubuntu-ports bionic-updates InRelease
4 packages can be upgraded. Run 'apt list --upgradable' to see them.
apt update complete.
Some packages can be upgraded. Please consider running sudo apt full-upgrade.
Creating an empty apt-package to install the necessary apt packages...
It will be named: pi-apps-51b4a14d
Depends: gobject-introspection, libgee-0.8-dev, libgirepository1.0-dev, libgnome-menu-3-dev, libgtk-3-dev, meson, valac
Installing the pi-apps-51b4a14d package...
The following NEW packages will be installed:
  pi-apps-51b4a14d
0 upgraded, 1 newly installed, 0 to remove and 4 not upgraded.
Get:1 /home/pi/pi-apps-51b4a14d.deb pi-apps-51b4a14d all 1.0 [800 B]
(Reading database ... 154795 files and directories currently installed.)
Unpacking pi-apps-51b4a14d (1.0) ...
Apt finished.
Package installation complete.
The Meson build system
Version: 0.56.1
Source dir: /home/pi/pikiss-gui
Build dir: /home/pi/pikiss-gui/build
Build type: native build
Project name: pikiss-gui
Project version: 0.0.8
C compiler for the host machine: cc (gcc 8.3.0 "cc (Raspbian 8.3.0-6+rpi1) 8.3.0")
C linker for the host machine: cc ld.bfd 2.31.1
Vala compiler for the host machine: valac (valac 0.42.5)
Host machine cpu family: arm
Host machine cpu: armv7l
Configuring config.h using configuration
Library config found: YES
Found pkg-config: /usr/bin/pkg-config (0.29)
Run-time dependency gio-unix-2.0 found: YES 2.58.3
Run-time dependency glib-2.0 found: YES 2.58.3
Run-time dependency gtk+-3.0 found: YES 3.24.5
Run-time dependency gdk-3.0 found: YES 3.24.5
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pango found: YES 1.42.3
Run-time dependency gee-0.8 found: YES 0.20.1
Run-time dependency libgnome-menu-3.0 found: YES 3.31.4
Library m found: YES
Program post_install.py found: YES (/usr/bin/python3 /home/pi/pikiss-gui/post_install.py)
Build targets in project: 1

Found ninja-1.10.1 at /usr/bin/ninja
ninja: Entering directory `build'
[1/3] Compiling Vala source ../src/Application.vala
[2/3] Compiling C object pikiss-gui.p/meson-generated_src_Application.c.o
[3/3] Linking target pikiss-gui
ninja: Entering directory `build'
[0/1] Installing files.

PiKISS Apps successfully synced.




Creating PiKISS GUI category dirs...
PiKISS GUI category dirs created successfully.

Creating PiKISS GUI menuentry...
PiKISS GUI menuentry created successfully.

Creating App import script...
App import script created successfully.

Creating PiKISS GUI Sync Apps menuentry...
PiKISS GUI Sync Apps menuentry created successfully.
For resyncing apps with PiKISS use menu launcher "PiKISS GUI Sync Apps"

Creating uninstall script
Uninstall script created successfully.

Syncing Apps with PiKISS...


Installing pikiss-gui to /home/pi/.local/bin
Installing /home/pi/pikiss-gui/data/pikiss-gui.png to /home/pi/.local/share/icons
Running custom install script '/usr/bin/python3 /home/pi/pikiss-gui/post_install.py'
Allowing packages required by the PiKISS GUI app to be uninstalled
These packages were: gobject-introspection, libgee-0.8-dev, libgirepository1.0-dev, libgnome-menu-3-dev, libgtk-3-dev, meson, valac
Purging the pi-apps-51b4a14d package...
The following packages will be REMOVED:
  pi-apps-51b4a14d*
0 upgraded, 0 newly installed, 1 to remove and 4 not upgraded.
                                                              (Reading database (Reading database ... 154795 files and directories currently installed.)
Removing pi-apps-51b4a14d (1.0) ...
Apt finished.
All packages have been purged succesfully.

Failed to install PiKISS GUI!
◢◣Need help? Copy the ENTIRE terminal output or take a screenshot.
Please ask on Github: https://github.com/Botspot/pi-apps/issues/new/choose
Or on Discord: https://discord.gg/RXSTvaUvuu
<details/>

**Comment by Jai-JAP** [comment]
@Botspot the best i can do to resolve this strange behaviour of install script is to change install script as below

```
#!/bin/bash

git_clone https://github.com/Jai-JAP/pikiss-gui || error 'Failed to clone PiKISS GUI repository!'

install_packages meson gobject-introspection libgee-0.8-dev libgirepository1.0-dev libgtk-3-dev valac libgnome-menu-3-dev || exit 1
"${DIRECTORY}/manage" install-if-not-installed piKiss || error "piKiss failed to install somehow!"

cd pikiss-gui
"${DIRECTORY}/etc/terminal-run" 'bash ./build.sh' 'Installing PiKISS-GUI' & 
while [ -e /proc/$! ]
  do
    sleep 1
    echo -en "\x1b[1;93mWaiting for child script to complete...\r\x1b[0m"
  done
```

**Comment by Jai-JAP** [comment]
Caught the error 
Redirections are not handled correctly
Removing ` &> >(tee -a "$logfile") ` from
https://github.com/Botspot/pi-apps/blob/b9aa4b4f24c3f620f05303518588607ef3bb1837/manage#L301
solves the issue

**Comment by Jai-JAP** [comment]
Here's the output of PiKISS GUI install as on screen after completion after removing redirectors
<details>
Installing PiKISS GUI...

Downloading pikiss-gui repository... Done
Will install these packages: meson gobject-introspection libgee-0.8-dev libgirepository1.0-dev libgtk-3-dev valac libgnome-menu-3-dev 
Running sudo apt update...
Hit:1 https://mcpi-revival.github.io/mcpi-packages buster InRelease
Hit:2 http://ppa.launchpad.net/apt-fast/stable/ubuntu focal InRelease
Hit:3 http://raspbian.raspberrypi.org/raspbian buster InRelease
Hit:4 http://ports.ubuntu.com/ubuntu-ports bionic-updates InRelease
Hit:5 http://archive.raspberrypi.org/debian buster InRelease
All packages are up to date.
apt update complete.
Creating an empty apt-package to install the necessary apt packages...
It will be named: pi-apps-51b4a14d
The pi-apps-51b4a14d package is already installed. Inheriting its dependencies: gobject-introspection, libgee-0.8-dev, libgirepository1.0-dev, libgnome-menu-3-dev, libgtk-3-dev, meson, valac
Depends: gobject-introspection, libgee-0.8-dev, libgirepository1.0-dev, libgnome-menu-3-dev, libgtk-3-dev, meson, valac
Installing the pi-apps-51b4a14d package...
pi-apps-51b4a14d is already the newest version (1.0).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Apt finished.
Package installation complete.
The Meson build system
Version: 0.56.1
Source dir: /home/pi/pikiss-gui
Build dir: /home/pi/pikiss-gui/build
Build type: native build
Project name: pikiss-gui
Project version: 0.0.8
C compiler for the host machine: cc (gcc 8.3.0 "cc (Raspbian 8.3.0-6+rpi1) 8.3.0")
C linker for the host machine: cc ld.bfd 2.31.1
Vala compiler for the host machine: valac (valac 0.42.5)
Host machine cpu family: arm
Host machine cpu: armv7l
Configuring config.h using configuration
Library config found: YES
Found pkg-config: /usr/bin/pkg-config (0.29)
Run-time dependency gio-unix-2.0 found: YES 2.58.3
Run-time dependency glib-2.0 found: YES 2.58.3
Run-time dependency gtk+-3.0 found: YES 3.24.5
Run-time dependency gdk-3.0 found: YES 3.24.5
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pango found: YES 1.42.3
Run-time dependency gee-0.8 found: YES 0.20.1
Run-time dependency libgnome-menu-3.0 found: YES 3.31.4
Library m found: YES
Program post_install.py found: YES (/usr/bin/python3 /home/pi/pikiss-gui/post_install.py)
Build targets in project: 1

Found ninja-1.10.1 at /usr/bin/ninja
ninja: Entering directory `build'
[3/3] Linking target pikiss-gui
ninja: Entering directory `build'
[0/1] Installing files.
Installing pikiss-gui to /home/pi/.local/bin
Installing /home/pi/pikiss-gui/data/pikiss-gui.png to /home/pi/.local/share/icons
Running custom install script '/usr/bin/python3 /home/pi/pikiss-gui/post_install.py'



Creating PiKISS GUI category dirs...
PiKISS GUI category dirs created successfully.

Creating PiKISS GUI menuentry...
PiKISS GUI menuentry created successfully.

Creating App import script...
App import script created successfully.

Creating PiKISS GUI Sync Apps menuentry...
PiKISS GUI Sync Apps menuentry created successfully.
For resyncing apps with PiKISS use menu launcher "PiKISS GUI Sync Apps"

Creating uninstall script
Uninstall script created successfully.

Syncing Apps with PiKISS...



PiKISS Apps successfully synced.


Installed PiKISS GUI successfully.

Closing in 30 seconds.
<details/>

**Comment by Jai-JAP** [comment]
I have a possible solution for this in theory.
I'll try to implement it tomorrow.

**Comment by Botspot** [comment]
I just tried using PiKISS-GUI and it was not fun! Clicking buttons did nothing, but I could see in task manager that various `apt` and `wget` processes were running.
Maybe make pikiss-gui open a terminal or something so the user can monitor progress.

**Comment by Jai-JAP** [comment]
> I just tried using PiKISS-GUI and it was not fun! Clicking buttons did nothing, but I could see in task manager that various `apt` and `wget` processes were running. Maybe make pikiss-gui open a terminal or something so the user can monitor progress.

In my testing i did open the terminal using x-terminal-emulator

**Comment by Jai-JAP** [comment]
@Botspot can you test the `test1` branch.
Now it should be fixed

**Comment by Jai-JAP** [comment]
Fixes are now availabe on master branch
Output fixes soon

**Comment by Jai-JAP** [comment]
@Botspot Ready to merge 
Output fixes working. Tested fine.

**Comment by Botspot** [comment]
Why is there still a countdown? Does it help the installation process somehow?
2022-02-25-113625_1920x1080_scrot

**Comment by Botspot** [comment]
Also PiKISS-GUI isn't working at all anymore.
https://user-images.githubusercontent.com/54716352/155763213-c4f458b9-7ef5-4754-836c-8b0a0dc767b1.mp4

**Comment by Jai-JAP** [comment]
It works now. That were a few typos.

I will remove countdown soon

**Pull request merged**